processor/src/main/java/org/mapstruct/ap/internal/model/source/TargetReference.java

```diff
@@ -138,8 +138,13 @@ public TargetReference build() {
                 return null;
             }
 
-
-            String[] segments = targetName.split( "\\." );
+            String targetNameTrimmed = targetName.trim();
+            if ( !targetName.equals( targetNameTrimmed ) ) {
+                messager.printMessage( method.getExecutable(), mapping.getMirror(), mapping.getSourceAnnotationValue(),
```

Comment:
This needs to be `mapping.getTargetAnnotationValue()` instead of `mapping.getSourceAnnotationValue()`. This property is actually not used in Java 8, but it is used correctly under Java 9, and not because of us, but because it was correctly implemented in the JDK only then.